build/api-docs-generator.js

```diff
@@ -1,562 +1,632 @@
- * maintaining the original directory structure.
- *
- * @param {string} dir - Current directory being processed
- * @param {string} baseDir - Base input directory for maintaining structure
+ * Creates required files & directories
+ * Writes content inside config.json & jsdoc.json
  */
-function processFilesRecursively(dir, baseDir) {
-    const files = fs.readdirSync(dir);
-
-    files.forEach(file => {
-        const filePath = path.join(dir, file);
-        const stats = fs.statSync(filePath);
-
-        if (stats.isDirectory()) {
-            // Create corresponding directory in output
-            const newOutputDir = path.join(outputDir, path.relative(baseDir, filePath));
-            if (!fs.existsSync(newOutputDir)) {
-                fs.mkdirSync(newOutputDir, { recursive: true });
-            }
-
-            // Recurse into subdirectories
-            processFilesRecursively(filePath, baseDir);
-        } else if (file.toLowerCase().endsWith('.js')) {
-            // Check if file should be included in API docs
-            if (fileContainsLine(filePath, '@INCLUDE_IN_API_DOCS')) {
-                const newFilePath = path.join(outputDir, path.relative(baseDir, filePath));
-                fs.copyFileSync(filePath, newFilePath);
-            }
-        }
-    });
+function createRequiredDir() {
+    fs.writeFileSync(
+        CONFIG_FILE,
+        JSON.stringify(CONFIG_JSON_TEMPLATE, null, 2)
+    );
+
+    fs.writeFileSync(
+        JSDOC_FILE,
+        JSON.stringify(JSDOC_JSON_TEMPLATE, null, 2)
+    );
+
+    fs.mkdirSync(MDX_FILES_DIR, { recursive: true });
 }
 
+
 /**
- * Removes RequireJS code and IIFE wrapper functions.
- * Also fixes some JSDoc issues that jsdoc-to-mdx doesn't handle well.
- * mdx doesn't allow {*} as a param value so modifying it to {any}
- * mdx doesn't allow {...} as a param value so modifying it to {rest}
- *
- *
- * @param {string} dir - Directory containing files to process
+ * To get all the JS files from src dir that are to be converted to MDX
+ * Creates a copy of all required JS files into JS-Files directory
+ * @returns {array} list of all required JS files
  */
-function removeRequireJSCode(dir) {
-    const files = fs.readdirSync(dir);
+function getJsFiles() {
+    // gets all JS files from src
+    // (even the ones that we don't need to add in API docs)
+    const files = glob.sync(`${SRC_DIR}/**/*.js`);
 
-    files.forEach(file => {
-        const filePath = path.join(dir, file);
-        const stats = fs.statSync(filePath);
+    // this gets all the files that we need to add in API docs
+    const requiredJSfiles = [];
+    for (const file of files) {
 
-        if (stats.isDirectory()) {
-            // Recurse into subdirectories
-            removeRequireJSCode(filePath);
-        } else {
-            try {
-                let modifiedContent = fs.readFileSync(filePath, 'utf8');
+        const content = fs.readFileSync(file, "utf-8");
```

Comment:
Check if its file file too here. we have a dir with name `/src/thirdparty/highlight.js` . So run the following commands before testing
1. `npm install` then run `npm run build` in phoenix repo to first build phoenix and its third party dependencies. 
2. Then do the development.